Create lfghelper

Lorkrakt opened the pull request:
The LFG Helper plugin helps RuneScape players quickly find groups for PVM content by providing a simple, user-friendly interface in the RuneLite sidebar. Players can specify group details like boss name, team size, clanchat, and world, and submit these details to a designated Discord channel via a webhook.

Features:
Simple Form Interface: Enter boss name, team size, clanchat, and more.
Discord Integration: Sends LFG request to a Discord channel with a custom webhook.
Role Support: Sends a message tagging the specified "PVMer" role on Discord.
Automatic Data Population: Includes the player’s in-game name in the request.

This plugin streamlines the process of looking for PVM groups, making it easier for players to coordinate and find groups within their communities

Lorkrakt commented:
This is ready for review whenever someone's ready.

LlemonDuck commented:
Required:
* Use an injected OkHttpClient to asynchronously send the discord payload instead of performing the i/o on swing thread

---

Recommended:
* Use ImageUtil#loadImageResource and ImageUtil#resizeImage to simplify your nav button image loading, optionally just not resizing at all since the icon is already 16x16
* Use `injector.getInstance(lfghelperpanel.class);` instead of manually instantiating the panel, which allows you to `@Inject` fields into it
* Use gson to build your discord payload which will prevent text-escaping issues

Lorkrakt commented:
Ready for review again.
Changes made:

Switched to OkHttpClient per RL Maintainer requirement
Made adaptions based on recommend

Fixed gson creation per workflow failure.

Lorkrakt commented:
Change made!
Ready to go again.
